### PR TITLE
Update node-xmpp-server & Node.js 4

### PR DESCRIPTION
**Pull request by sonnyp**
* node-xmpp-core is not needed (and possibly harmful as you possibly could be using a different version as node-xmpp-server)
* node-xmpp-server received several important fixes
* node-xmpp-server now installs on Node.js 4

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,3 +5,17 @@ language: node_js
...
+env:
+  - CXX="g++-4.8"
+
+addons:
+  apt:
+    sources:
+    - ubuntu-toolchain-r-test
+    packages:
+    - g++-4.8
+    - gcc-4.8
```
Comment (sonnyp) [inline .travis.yml]: required for module compilation on travis/node 4

File: package.json [hunk 2]
```diff
@@ -81,8 +85,7 @@
     "mongoose-validate": "0.0.5",
     "multer": "^0.1.8",
     "node-uuid": "^1.4.3",
-    "node-xmpp-core": "2.0.0",
-    "node-xmpp-server": "1.0.2",
+    "node-xmpp-server": "2.1.0",
```
Comment (sonnyp) [inline package.json]: any reason why the version is pinned while other dependencies aren't?

**Comment by mitzip** [comment]
great job!

File: package.json (hunk 2 above)
Comment (hhaidar) [inline package.json]: Yea probably doesn't have to be pinned anymore

**Comment by hhaidar** [comment]
Thank you so much!

**Pull request merged**